chef-utils/lib/chef-utils/dsl/render_helpers.rb

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../internal"
+require_relative "../toml"
+
+require 'json'
+require 'yaml'
+
+module ChefUtils
+  module DSL
+    module RenderHelpers
+      include Internal
+
+      # pretty-print a hash as a JSON string
+      def render_json(hash)
+        JSON.pretty_generate(hash) + "\n"
+      end
+
+      # pretty-print a hash as a TOML string
+      def render_toml(hash)
+        ChefUtils::Toml::Dumper.new(hash).toml_str
+      end
+
+      # pretty-print a hash as a YAML string
+      def render_yaml(hash)
+        yaml_content = hash.transform_keys(&:to_s).to_yaml
+        #above replaces first-level keys with strings, below the rest
+        yaml_content.gsub!(' :',' ')
```

Comment:
This does not seem like a safe thing to do on arbitrary data. In ruby it is straighforward to recursively convert a hash's keys to strings though:

```
def render_yaml(hash)
  converter = lambda do |object|
    case object
    when Hash
      object.transform_keys(&:to_s).transform_values(&converter)
    else
      object
    end
  end

  converter.call(hash).to_yaml
end
```

Reply:
probably needs a `when Array` that recursively calls the converter on all the elements.

i think the api probably needs to take an optional flag to turn this convert off as well in case users want to actually render symbols.